docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-6.0-to-2024.2/upgrade-guide-from-6.0-to-2024.2-generic.rst

```diff
@@ -0,0 +1,401 @@
+.. _SCYLLA_RPM_SRC_REPO: https://www.scylladb.com/download/?platform=centos&version=scylla-6.0
+
+.. |DEBIAN_NEW_REPO| replace:: Debian
+.. _DEBIAN_NEW_REPO: https://www.scylladb.com/customer-portal/?product=ent&platform=debian-11&version=stable-release-2024.2
+
+.. |UBUNTU_NEW_REPO| replace:: Ubuntu
+.. _UBUNTU_NEW_REPO: https://www.scylladb.com/customer-portal/?product=ent&platform=ubuntu-22.04&version=stable-release-2024.2
+
+.. |SCYLLA_DEB_NEW_REPO| replace:: ScyllaDB deb repo (|DEBIAN_NEW_REPO|_, |UBUNTU_NEW_REPO|_)
+
+.. |SCYLLA_RPM_NEW_REPO| replace:: ScyllaDB rpm repo
+.. _SCYLLA_RPM_NEW_REPO: https://www.scylladb.com/customer-portal/?product=ent&platform=centos7&version=stable-release-2024.2
+
+.. |ROLLBACK| replace:: rollback
+.. _ROLLBACK: ./#rollback-procedure
+
+.. |SCYLLA_METRICS| replace:: ScyllaDB Enterprise Metrics Update - ScyllaDB Enterprise 6.0 to 2024.2
+.. _SCYLLA_METRICS: ../metric-update-6.0-to-2024.2
+
+=============================================================================
+Upgrade Guide - |SCYLLA_NAME| |SRC_VERSION| to |NEW_VERSION|
+=============================================================================
+
+This document is a step-by-step procedure for upgrading from |SCYLLA_NAME| |SRC_VERSION| 
+to |SCYLLA_NAME| Enterpise |NEW_VERSION|, and rollback to version |SRC_VERSION| if required.
+
+This guide covers upgrading ScyllaDB on Red Hat Enterprise Linux (RHEL) CentOS, Debian, 
+and Ubuntu. See :doc:`OS Support by Platform and Version </getting-started/os-support>` 
+for information about supported versions.
+
+This guide also applies when you're upgrading ScyllaDB Enterprise official image on EC2, 
+GCP, or Azure.
+
+
+Before You Upgrade ScyllaDB
+================================
+
+**Upgrade Your Driver**
+
+If you're using a :doc:`ScyllaDB driver </using-scylla/drivers/cql-drivers/index>`, 
+upgrade the driver before you upgrade ScyllaDB. The latest two versions of each driver 
+are supported.
+
+**Upgrade ScyllaDB Monitoring Stack**
+
+If you're using the ScyllaDB Monitoring Stack, verify that your Monitoring Stack 
+version supports the ScyllaDB version to which you want to upgrade. See 
+`ScyllaDB Monitoring Stack Support Matrix <https://monitoring.docs.scylladb.com/stable/reference/matrix.html>`_.
+  
+We recommend upgrading the Monitoring Stack to the latest version.
+
+**Check Feature Updates**
+
+See the ScyllaDB Release Notes for the latest updates. The Release Notes are published 
+at the `ScyllaDB Community Forum <https://forum.scylladb.com/>`_.
+
+.. note::
+
+   Unlike ScyllaDB 6.0, ScyllaDB Enterprise 2024.2 has **tablets disabled by default**
+   on the keyspace level. This means that after you upgrade to 2024.2:
+
+   * Tables in keyspaces created with default settings in version 6.0 will
+     have tablets enabled.
+   * Tables in keyspaces created with default settings after upgrading to
+     version 2024.2 will have tablets disabled.
```

Comment:
@bhalevy Does this sound right to you? I checked the code and it seems to me that as long as tablets feature is enabled, we will default to using tablets. So if 6.0 had tablets enabled then after the upgrade we will inherit the behavior of using tablets by default even if keyspace has default settings.

Code (2024.2):
```
ks_prop_defs::init_tablets_options ks_prop_defs::get_initial_tablets(const sstring& strategy_class, bool enabled_by_default) const {
    // FIXME -- this should be ignored somehow else
    init_tablets_options ret{ .enabled = false, .specified_count = std::nullopt };
    if (locator::abstract_replication_strategy::to_qualified_class_name(strategy_class) != "org.apache.cassandra.locator.NetworkTopologyStrategy") {
        return ret;
    }

    auto tablets_options = get_map(KW_TABLETS);
    if (!tablets_options) {
        return enabled_by_default ? init_tablets_options{ .enabled = true } : ret;
    }

...

lw_shared_ptr<data_dictionary::keyspace_metadata> ks_prop_defs::as_ks_metadata(sstring ks_name, const locator::token_metadata& tm, const gms::feature_service& feat) {
    auto sc = get_replication_strategy_class().value();
    auto initial_tablets = get_initial_tablets(sc, feat.tablets);
    // if tablets options have not been specified, but tablets are globally enabled, set the value to 0

```

Reply:
@xemul Please review as well, as you worked on this.

Reply:
> Tablets should not be the default for 2024.2 Discussion should move to the Enterprise repo.

Is it fine if they're default after upgrade from 6.0 which had them enabled?

Reply:
> Tables in keyspaces created with default settings after upgrading to version 2024.2 will have tablets disabled.

No.  That's not how it is implemented in 2024.2.
Tablet support is determined by a configuration option and if it was enabled with 6.0 then it will remain enabled also with 2024.2.
Only after explicitly setting it to false (or commenting it out from scylla.yaml), tablets will get disabled (for new keyspaces).

Reply:
@bhalevy, in this case, we need to change the implementation (core or config)